src/lib/transaction.ts

```diff
@@ -736,3 +456,327 @@ export async function sendTransaction(
     ...sendOptions,
   });
 }
+
+/**
+ * Sends a versioned transaction with compute unit optimization and automatic retries
+ *
+ * @param connection - The Solana connection object
+ * @param instructions - Array of instructions to include in the transaction
+ * @param signers - Array of signers needed for the transaction
+ * @param priorityFee - Priority fee in microLamports (default: 10000 which is the minimum required for helius to see a transaction as priority)
+ * @param lookupTables - Optional array of address lookup tables for account compression
+ * @param options - Optional configuration for retry mechanism and compute units
+ * @returns Promise that resolves to the transaction signature
+ *
+ * @remarks
+ * This function:
+ * 1. Automatically calculates and adds compute unit instructions if not present
+ * 2. Creates a v0 transaction message with the provided instructions
+ * 3. Signs and sends the transaction with automatic retries
+ * 4. Provides status updates through the callback
+ *
+ * Status updates include:
+ * - "computeUnitBufferAdded": Compute unit instructions were added
+ * - "created": Transaction was created
+ * - "signed": Transaction was signed
+ * - "sent": Transaction was sent (includes signature)
+ * - "confirmed": Transaction was confirmed
+ *
+ * @example
+ * ```typescript
+ * const signature = await sendVersionedTransaction(
+ *   connection,
+ *   instructions,
+ *   [payer],
+ *   10000,
+ *   lookupTables,
+ *   {
+ *     computeUnitBuffer: { multiplier: 1.1 },
+ *     onStatusUpdate: (status) => console.log(status),
+ *   }
+ * );
+ * ```
+ */
+export async function sendVersionedTransaction(
```

Comment:
why is this a separate function? instead of a single `sendTransaction` function? there is really nothing specific to sending version transactions in here.

please just add version transaction support to the existing sending function

Reply:
The signature is different. This one takes an array of instructions so that i can still add blockhash and sign and also add the lookupTables. 
Transaction and VersionedTransaction are different types and work a bit differently thats why i made it two sperate functions. But i managed to refactor it a bit to have less duplicated code for the sending.